guide/interface-components/components.md

```diff
@@ -0,0 +1,316 @@
+			.setStyle(ButtonStyle.Secondary);
+
+		const confirm = new ButtonBuilder()
+			.setCustomId('confirm')
+			.setLabel('Confirm')
+			.setStyle(ButtonStyle.Danger);
+	},
+};
+```
+
+::: tip
+There are UX principles in play when creating buttons! We've created a Secondary (grey) Cancel button for the less important action, and a Danger (red) Confirm button for the "destructive" ban action!
+:::
+
+In order to send any type of message component, they must be placed in an `ActionRow`. Each row can contain up to 5 buttons or 1 select menu, up to a total of 5 rows in a message. To create one, we'll use the <DocsLink path="class/ActionRowBuilder"/> class, add our buttons to it, then send it in the `InteractionReplyOptions#components` array.
+
+```js {1,18-19,21-25}
+const { ActionRowBuilder, ButtonBuilder, ButtonStyle } = require('discord.js');
+
+module.exports = {
+	// data: new SlashCommandBuilder()...
+	async execute(interaction) {
+		const target = interaction.options.getUser('target');
+
+		const cancel = new ButtonBuilder()
+			.setCustomId('cancel')
+			.setLabel('Cancel')
+			.setStyle(ButtonStyle.Secondary);
+
+		const confirm = new ButtonBuilder()
+			.setCustomId('confirm')
+			.setLabel('Confirm')
+			.setStyle(ButtonStyle.Danger);
+
+		const row = new ActionRowBuilder()
+			.addComponents(cancel, confirm);
+
+		interaction.reply({
+			content: `Are you sure you want to ban ${target.username}?`,
+			components: [row],
+		});
+	},
+};
+```
+
+Restart your bot and test your ban command. If all goes well, you should see something like this.
+
+<DiscordMessages>
+	<DiscordMessage profile="bot">
+		<template #interactions>
+			<DiscordInteraction profile="user" :command="true">button</DiscordInteraction>
+		</template>
+		Are you sure you want to ban DiscordTroll69?
+		<template #actions>
+			<DiscordButtons>
+				<DiscordButton type="secondary">Cancel</DiscordButton>
+			</DiscordButtons>
+			<DiscordButtons>
+				<DiscordButton type="danger">Confirm</DiscordButton>
+			</DiscordButtons>
+		</template>
+	</DiscordMessage>
+</DiscordMessages>
+
+::: warning
+If you're using TypeScript you'll need to specify the type of components your action row holds. This can be done by specifying the component builder you will add to it using a generic parameter in <DocsLink path="class/ActionRowBuilder"/>.
+
+```diff
+- new ActionRowBuilder()
++ new ActionRowBuilder<ButtonBuilder>()
+```
+:::
+
+## Receiving button interactions
+
+Our buttons created and shown to the user, but there's no functionality behind them yet. For this confirmation workflow, your bot is expecting a single `ButtonInteraction` to be received before it can continue. The best way to handle this is with the <DocsLink path="class/Message?scrollTo?=awaitMessageComponent" /> method. This method wraps a Promise around an `InteractionCollector` to return only the first interaction that meets the criteria - checkout the [full guide on Collectors]() for more information on those.
```

Comment:
Our -> We
Also needs some rephrasing, this reads super weird as a half-sentence is followed by a "but"
"checkout" -> "check out"